src/main/java/org/mockito/internal/framework/DisabledMockException.java

```diff
@@ -0,0 +1,33 @@
+package org.mockito.internal.framework;
+
+import org.mockito.invocation.Invocation;
+import org.mockito.invocation.InvocationContainer;
+import org.mockito.invocation.MockHandler;
+import org.mockito.mock.MockCreationSettings;
+
+/**
+ * Thrown when a mock is accessed after it has been disabled by
+ * {@link org.mockito.MockitoFramework#disableInlineMocks(String)}.
+ */
+public class DisabledMockException extends RuntimeException {
+    public DisabledMockException() {
+        super("Mock accessed after inline mocks were cleared");
+    }
+
+    public static MockHandler HANDLER = new MockHandler() {
```

Comment:
Nit: since the `MockHandler` throws the exception, I think this should be reversed. E.g. `public class DisabledMockHandler implements MockHandler {}` and the `DisabledMockException` should be the inner class. Then we can still have `public static DisabledMockHandler HANDLER = new DisabledMockHandler();`

Another nit: `DisabledMockException extends MockitoException` as we mark all our exceptions as `MockitoException`